tests/test_add.py

```diff
@@ -40,13 +39,32 @@ def test_add(self, mocker: Any) -> None:
 
         self.repo_sandbox.new_branch("bugfix/feature_fail")
 
+        # Test `git machete diff` without providing the branch name
         assert_command(
-            ['add', '-y', 'bugfix/feature_fail'],
+            ['add', '-y'],
             'Adding `bugfix/feature_fail` onto the inferred upstream (parent) branch `develop`\n'
             'Added branch `bugfix/feature_fail` onto `develop`\n',
             strip_indentation=False
         )
 
+        self.repo_sandbox.check_out('develop')
+        self.repo_sandbox.new_branch("bugfix/some_feature")
+        assert_command(
+            ['add', '-y', 'bugfix/some_feature'],
+            'Adding `bugfix/some_feature` onto the inferred upstream (parent) branch `develop`\n'
+            'Added branch `bugfix/some_feature` onto `develop`\n',
+            strip_indentation=False
+        )
+
+        self.repo_sandbox.check_out('develop')
+        self.repo_sandbox.new_branch("bugfix/another_feature")
+        assert_command(
+            ['add', '-y', 'refs/heads/bugfix/another_feature'],
```

Comment:
Gr8 🎉 curious how it affected the overall coverage... https://app.codecov.io/gh/VirtusLab/git-machete/branch/bug%2F497-parsing-branch-name doesn't seem to display the most recent data here (probably since the tests haven't been passing recently; failed tests don't count for coverage)

Reply:
(btw do you have access to Codecov?)

Reply:
@PawelLipski yes, i do have access to Codecov but as you pointed out it doesn't display the most recent data, will come back to it once the tests pass